### DISCUSSION TITLE
Ky alternative for axios interceptors

### DISCUSSION THREAD

**Discussion by clicktodev** [sindresorhus/ky#363] [Q&A] [answered]
Something like axios interceptors is very important to the project I'm currently working on.
Does ky have something similar?

**Comment by sindresorhus** [comment, marked as answer]
Yes, but Ky calls them [hooks](https://github.com/sindresorhus/ky#hooks).

**Reply by clicktodev** [reply]
Great, and is it possible to define them once globally for each request that is made using ky?
Because in Axios i can define an interceptor in my single axios instance at the root of my project and it will run for each request i make in the entire application.
And if I wanna ignore the interceptor for a specific request, i can just define a new axios instance for that request.

**Reply by sindresorhus** [reply]
> Great, and is it possible to define them once globally for each request that is made using ky?

No. That's an intional choice. Global hooks (interceptors) is a bad idea for so many reasons.

Instead, you can define the hooks in separate file using `ky.create` and the import that file everywhere in your app instead of importing `ky`.

If you want to ignore the hook for a specific request, you can use `ky.extend` on your previously created instance.

https://github.com/sindresorhus/ky#kycreatedefaultoptions

**Reply by clicktodev** [reply]
Thanks for the detailed response, You're doing Great work with the library 👏
